core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/BackupDatalakeDatabaseFlowEventChainFactory.java

```diff
@@ -32,8 +40,11 @@ public String initEvent() {
     @Override
     public FlowTriggerEventQueue createFlowTriggerEventQueue(DatabaseBackupTriggerEvent event) {
         Queue<Selectable> flowEventChain = new ConcurrentLinkedQueue<>();
-        flowEventChain.add(new StackEvent(SaltUpdateEvent.SALT_UPDATE_EVENT.event(), event.getResourceId(), event.accepted()));
-        flowEventChain.add(new DatabaseBackupTriggerEvent(DATABASE_BACKUP_EVENT.event(), event.getResourceId(),
+        StackStatus stackStatus = stackService.getCurrentStatusByStackId(event.getResourceId());
+        if (!stackStatus.getDetailedStackStatus().equals(DETERMINE_DATALAKE_DATA_SIZES_FINISHED)) {
```

Comment:
is this the optimization not to invoke salt update too often?

Reply:
Yes it is, with this the DB backup will not invoke the salt update if the determine datalake data sizes flow was already run.